netlify/netlify-plugin-nextjs
# feat: add experimental support for TTL

### Summary

Adds experimental support for ttl in pages with revalidate set. Parses the cache-control header returned by Next and extracts the `s-maxage` value, which it then uses in the ttl response. This isnlt enabled by default, but requires an env var flag.

As part of the build process we need to patch the `next-server.js` file, because we need to disable caching for ISR pages and currently this is only supported for `minimalMode`. We do this by adding another env var to the line where it checks for all the conditions for disabling the SSG cache.

### Test plan

1. Visit [a page with revalidate set](https://deploy-preview-833--netlify-plugin-nextjs-demo.netlify.app/getStaticProps/withRevalidate/1/). The rendered-at time will probably be old. If the rendered-at time is the current wall time then it's a fresh deploy so wait a minute for it to be stale then load again.
2. Wait for 1 minute for the revalidated content to propagate to your edge node
3. Load the page again. The rendered-at time should match the time at step 1 (so about a minute in the past)

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal

![image](https://user-images.githubusercontent.com/213306/143423357-431eff7d-401d-489c-9473-42abbb1a533e.png)
